.gitignore

```diff
@@ -23,6 +38,7 @@
 .idea/
 *.recipe
 *.idb
+*.iobj
```

Comment:
This adds some common ignore patterns. For us it concerns the build outputs, for example when a build script or Makefile is run. It will stop files like `build/Linux/Clang/libwebui-2-static-x64.so` being tracked as changes by a users git version control when the mentioned files are executed.

The added patterns are common ignore patterns and to keep a little better overview I'm grouping them with the present patterns into categories like it is done e.g. in the [gitignore template](https://github.com/github/gitignore/blob/main/C.gitignore) for C files.

One thing I want to mention is that we still have static / shared library files in the text-editor example. They won't be ignored, since they are already part of repository and will not be deleted with this PR. Maybe we should also create a Makefile for the text-editor example, like it is done for the other examples so we can also remove them? But this would be out of the scope of this PR and should be done separately I think.